modules/external_petsc_solver/src/base/ExternalPetscSolverApp.C

```diff
@@ -48,6 +58,28 @@ ExternalPetscSolverApp::registerAll(Factory & f, ActionFactory & af, Syntax & /*
   /* register custom execute flags, action syntax, etc. here */
 }
 
+std::shared_ptr<Backup>
+ExternalPetscSolverApp::backup()
+{
+  auto backup = MooseApp::backup();
+
+  PetscVector<Number> petsc_sol(_petsc_sol, comm());
```

Comment:
What's the reason for this copy construction? I'm sure you have one.

Reply:
Just want to temporally add a wrapper to `_petsc_sol` (PETSc Vec), and we can use moose routines.  This is a not  copy constructor.  We construct a `PetscVector<Number>` using a `Vec` as input

Reply:
Ok duh, forgot that `_petsc_sol` was just a `Vec`. 👍